### PR TITLE
CU-866arq3x9 - contact page and integration with hubspot form

### PR DESCRIPTION
**Pull request by vaionexbusra**
The app file that comes with the shadcn has to be deleted.

### CODE REVIEW

File: src/app/(content)/contact/page.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import ContactComponent from '@/components/Contact'
+
+export default function Login() {
```
Comment (paulsp94) [inline src/app/(content)/contact/page.js:3]: This should not be named login

File: src/app/api/contactInfo/route.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import axios from 'axios'
+
+export async function POST(request) {
+  const body = await request.json()
+  const referer = request.headers.get('referer')
+
+  const formFields = {
```
Comment (paulsp94) [inline src/app/api/contactInfo/route.js:7]: you can define this outside the component

File: src/app/api/contactInfo/route.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        name: 'message',
+        value: body.message,
+      },
+    ],
+    context: {
+      pageUri: 'https://vbs-mu.vercel.app/',
+    },
+  }
+
+  console.log('Form Fields:', formFields)
```
Comment (paulsp94) [inline src/app/api/contactInfo/route.js:35]: remove the log

File: src/components/Contact/index.js [hunk 4]
```diff
@@ -0,0 +1,236 @@
+'use client'
+import React, { useState } from 'react'
+import { BuildingOffice2Icon, EnvelopeIcon } from '@heroicons/react/24/outline'
+import { toast } from 'react-toastify'
+
+const InputField = ({
```
Comment (paulsp94) [inline src/components/Contact/index.js:6]: lets put that in their own file

Comment (paulsp94) [reply src/components/Contact/index.js:6]: Or look into using https://ui.shadcn.com/docs/components/input

**Review by paulsp94 — CHANGES_REQUESTED** [review]

File: src/app/api/contactInfo/route.js (hunk 2 above)
Comment (vaionexbusra) [reply src/app/api/contactInfo/route.js:7]: Isn't it already in the route.js?

File: src/components/Contact/index.js [hunk 5]
```diff
@@ -2,6 +2,9 @@
 import React, { useState } from 'react'
 import { toast } from 'react-toastify'
 import Image from 'next/image'
+import { Input } from 'shadcn-ui/input' // Assuming shadcn-ui provides an Input component
```
Comment (paulsp94) [inline src/components/Contact/index.js:5]: the input is in src/components/ui/input.jsx

File: src/components/Contact/index.js [hunk 6]
```diff
@@ -54,6 +57,44 @@ export default function ContactComponent() {
       toast.error('An error occurred. Please try again.')
     }
   }
+
+  const formConfig = [
```
Comment (paulsp94) [inline src/components/Contact/index.js:61]: you can pull that outside of the component it has no dependency on the data inside of the component

File: src/components/Contact/index.js [hunk 7]
```diff
@@ -142,111 +183,48 @@ export default function ContactComponent() {
...
+                <div className="mt-2.5" key={field.name}>
+                  <Label
+                    className="block text-sm font-semibold leading-6 text-gray-900"
+                    htmlFor={field.name}
+                  >
+                    {field.label}
+                  </Label>
+                  <div className="mt-2.5">
+                    {field.isTextarea ? (
+                      <textarea
```
Comment (paulsp94) [inline src/components/Contact/index.js:196]: https://ui.shadcn.com/docs/components/textarea

we could also go for their textarea

File: src/components/Contact/index.js [hunk 8]
```diff
@@ -142,111 +183,48 @@ export default function ContactComponent() {
...
+                  </div>
                 </div>
-              </div>
+              ))}
             </div>
             <div className="mt-8 flex justify-end">
-              <button
+              <Button
                 type="submit"
                 className="rounded-md bg-gradient-to-br from-gray-900 to-gray-600 px-4 py-2 text-sm font-semibold text-white shadow-sm hover:bg-opacity-75 "
```
Comment (paulsp94) [inline src/components/Contact/index.js:223]: if we use the button from shadcn do we still need all of those styles?

Comment (paulsp94) [reply src/components/Contact/index.js:223]: check the same for label and input

File: src/components/Contact/index.js (hunk 5 above)
Comment (paulsp94) [reply src/components/Contact/index.js:5]: same for the other two components, this cant work did you tested it?

**Review by paulsp94 — CHANGES_REQUESTED** [review]

File: package.json [hunk 9]
```diff
@@ -22,6 +24,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-toastify": "^9.1.3",
+    "shadcn-ui": "^0.3.0",
```
Comment (paulsp94) [inline package.json:27]: that does not need to be added

File: src/app/(content)/about/page.js [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { Hero } from '@/components/Layout'
+import { Hero } from '@./components/Layout'
```
Comment (paulsp94) [inline src/app/(content)/about/page.js:1]: that looks wrong

**Pull request closed without merging**